addons/account/models/account_invoiceable_order.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command, _, api, fields, models
+from odoo.tools import float_is_zero
+
+
+class InvoiceableOrder(models.AbstractModel):
+    _name = 'account.invoiceable.order'
```

Comment:
From my experience in Logistics (purchase scope) there are a lot of cases where we have overlapping functionality and are maintaining/fixing a lot of similar stuff between sales and purchase. As an example, my original PR was a full copy of all of the down payment stuff from sales. That's why I personally think the purpose of these classes is to build a good foundation for adding more overlapping sale/purchase code in the future and to maintain it in a single place instead of in parallel. So I think the names InvoiceableOrder and InvoiceableLine are quite good.

Reply:
Agreed after reading  https://github.com/odoo/odoo/pull/144608#pullrequestreview-1829400123 but why not keeping the mixin keyword ? It'd be much easier to understand that this is not an object on its own.

Reply:
![image](https://github.com/odoo/odoo/assets/3251665/c66446ec-85d6-4ef6-a144-79b353f247aa)